### PR TITLE
fix: replace colors to chalk

### PR DESCRIPTION
**Pull request by Samridhi-98**
**Description**

-  Added package chalk in `template/package.json`
-  Removed package colors from  `template/package.json`
-  Updated  respective code in `src/api/middleware/...` & `src/api/index.js`

**Related issue(s)**

**Fixes** #110

### CODE REVIEW

**Comment by Samridhi-98** [comment]
@derberg  Please review my PR.

**Review by derberg — CHANGES_REQUESTED** [review]
that was super fast 😄 

you need to do fixes to `template/package.json` because it is a templated file that cannot be formatted the way autoformatter did it for you.

for example `{%` cannot become `{ %`

or `{% if asyncapi.server(params.server).protocol() === 'kafka' or asyncapi.server(params.server).protocol() === 'kafka-secure' -%}` cannot become

```
        { % -endif - % } { %
            if asyncapi.server(params.server).protocol() === 'kafka'
            or asyncapi.server(params.server).protocol() === 'kafka-secure' - %
        }
```

These changes generate a broken package.json.

Please make sure `package.json` has no formatting changes, only dependencies.

Then please test it locally to make sure it works. Please use this instructions -> https://github.com/asyncapi/nodejs-template#cli

Just instead of `ag https://bit.ly/asyncapi @asyncapi/nodejs-template -o output -p server=production` do `ag https://bit.ly/asyncapi ./ -o output -p server=production` from the root of cloned repo

**Comment by Samridhi-98** [comment]
> 

Sure, I will do the mentioned changes.

**Comment by Samridhi-98** [comment]
@derberg After disabling autoformatter I got so many errors in `index.js` and `package.json` and also unable to test the exchanges please help me out.

**Screenshot**

image

**Comment by derberg** [comment]
@Samridhi-98 errors are normal, unfortunately. The think is that these files are "templates", so files with Nunjucks templating injected, as depending on your AsyncAPI file that you provide as input, the generator generates different output. 

For example, [this](https://github.com/asyncapi/nodejs-template/blob/master/template/package.json#L14-L16) part means that `hermesjs-mqtt` will be included in package.json only for https://raw.githubusercontent.com/asyncapi/asyncapi/2.0.0/examples/2.0.0/streetlights.yml file because protocol used there is mqtt: `protocol: mqtt`

This is why all these files look so broken. We just need to find time to rewrite this template from Nunjucks to React engine.

So:
- you for sure have to remove from this PR `package-lock.json` 
- tell me what problems do you have with testing, the instructions that I provided are wrong or?

**Comment by Samridhi-98** [comment]
> 

Yeah, I got that template thing and understand the logic as well but the problem is I have implemented the changes that you suggested but now due to template layout error as shown in the above screenshot I am unable to test it locally.

I tried executing this command  `ag https://bit.ly/asyncapi ./ -o output -p server=production`  and got the error mentioned in the screenshot above and while executing this command  `ag https://bit.ly/asyncapi @asyncapi/nodejs-template -o output -p server=production`  I got something like this mentioned in the screenshot below.

image

So, I just don't understand now how to test the implementations.

**Comment by derberg** [comment]
@Samridhi-98 ah sorry, I looked at editor but overlooked the console, sorry about that 🤦🏼 

So yeah, you have an error there, incorrect syntax, there should be space after `and`

```
{%- if params.securityScheme and (asyncapi.server(params.server).protocol() === 'kafka' or asyncapi.server(params.server).protocol() === 'kafka-secure') and asyncapi.components().securityScheme(params.securityScheme).type() === 'X509' %}
```

It is strange for me though, as I see in PR that you used autoformatter in `package.json` only, not `index.js`, the part where you get error was not changed in your PR 🤔 

Also `--debug` flag sometimes provides more help, not always though

**Comment by Samridhi-98** [comment]
> @Samridhi-98 ah sorry, I looked at editor but overlooked the console, sorry about that 🤦🏼
> 
> So yeah, you have an error there, incorrect syntax, there should be space after `and`
> 
> ```
> {%- if params.securityScheme and (asyncapi.server(params.server).protocol() === 'kafka' or asyncapi.server(params.server).protocol() === 'kafka-secure') and asyncapi.components().securityScheme(params.securityScheme).type() === 'X509' %}
> ```
> 
> It is strange for me though, as I see in PR that you used autoformatter in `package.json` only, not `index.js`, the part where you get error was not changed in your PR 🤔
> 
> Also `--debug` flag sometimes provides more help, not always though

Actually I am getting errors while executing this cmd `ag https://bit.ly/asyncapi ./ -o output -p server=production` .

And if I execute this  `ag https://bit.ly/asyncapi @asyncapi/nodejs-template -o output -p server=production`  it gave the output for master, not for my local changes.

**Comment by derberg** [comment]
Yes, because `@asyncapi/nodejs-template` means that latest release will be used, and `./` points to your local template. So release of course works, and something is wrong with your changes. Just fic the spacing issue I wrote about, and then run again with and without debug flag and share logs please

**Comment by Samridhi-98** [comment]
> ag https://bit.ly/asyncapi ./ -o output -p server=production

I am getting this error while running `ag https://bit.ly/asyncapi ./ -o output -p server=production`, Also I fixed all broken syntax as you suggested to do.

image

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 node_modules
 output
 test/temp
+template/package-lock.json
```
Comment (derberg) [inline .gitignore:5]: it is not necessary to ignore it

**Comment by Samridhi-98** [comment]
> Oh, I saw this issue someone reported in Slack yesterday. I was super confused but now I think I know what happened.
> 
> So the issue is caused by this file https://github.com/alexei/sprintf.js/blob/master/demo/angular.html that has some different templating. But why would the template care about this file? it should not, it should not get into render engine?
> 
> My only guess is that you probably ran `npm i` inside `template` directory, and now in your local, `template` dir contains also entire `node_modules`. So when you run `ag`, it feeds generator with entire `node_module`.
> 
> Solution: just remove `template/node_modules`. The only installation you should make on local, when developing, is in the root of the project

I fixed the issue and now it's working, also tested it locally as well.

I am closing this PR and creating the fresh one, extremely sorry for troubling.

**Pull request closed without merging**